### PR TITLE
🎉 New source: US census

### LINKED ISSUE #3662: Add a US Census connector
## Tell us about the problem you're trying to solve
I am trying to ingest public census data as part of a POC: https://api.census.gov/data/timeseries/eits/mrts/examples.html

But their responses are non standard: https://www.census.gov/data/developers/guidance/api-user-guide.Core_Concepts.html

image (3)

e.g.

```js
[
  ["column_a", "column_b", "column_c"],
  ["value_a1", "value_b1", "value_c1"],
  ["value_a2", "value_b2", "value_c2"]
]
```

## Describe the solution you’d like
I would like Airbyte to be able to ingest this data, currently we try to use the HTTP connector and we get:

```
E   pydantic.error_wrappers.ValidationError: 1 validation error for AirbyteRecordMessage
E   data
E     value is not a valid dict (type=type_error.dict)
```

I see 3 potential solutions (thanks marcosmarxm for the idea to build something census specific):

1. Create a US Census specific connector, which supports their non-conventional responses
1. Patch the HTTP connector: Airbyte recognizes "list of list" as a special response format, assumes the first row is the column names, and returns something like:

```js
{"data": [
  {"column_a": "value_a1", "column_b": "value_b1", "column_c": "value_c1"},
  {"column_a": "value_a2", "column_b": "value_b2", "column_c": "value_c2"},
]}
```

1. OR patch the http connector: do not recognize it as a special format, and return

```js
[
  {"data": ["column_a", "column_b", "column_c"]},
  {"data": ["value_a1", "value_b1", "value_c1"]},
  {"data": ["value_a2", "value_b2", "value_c2"]}
]
```

## Are you willing to submit a PR?
Yes, I opened a PR that patches the HTTP connector, but I would like to attempt creating a census specific connector instead.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200391070058960) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by dmateusp**
## What
Resolves https://github.com/airbytehq/airbyte/issues/3662

Adds a source specific to the US Census API which gives responses in a non-standard format which was not handled nicely by the existing HTTP source

## How

Parses the "list of lists of string" response by taking the first list as the "header" and turning the following lists into records.
Effectively the first list makes up the schema of the data.

The schema is discovered dynamically by reading the header and first record of the response. All data types are strings as the US Census API only returns strings.

## Recommended reading order
1. [Parsing the response](https://github.com/airbytehq/airbyte/pull/4228/files#diff-bab5901604a527c3ed277a1c4be586650291dded9deb31431cd03d4bc2959ed0R82)
2. [Getting the schema dynamically](https://github.com/airbytehq/airbyte/pull/4228/files#diff-bab5901604a527c3ed277a1c4be586650291dded9deb31431cd03d4bc2959ed0R167)
3. [Sample configuration](airbyte-integrations/connectors/source-us-census/integration_tests/sample_config.json)
4. Unit tests: airbyte-integrations/connectors/source-us-census/unit_tests/unit_test.py

## Pre-merge Checklist

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Secrets are annotated with `airbyte_secret` in the connector's spec
- [x] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] `README.md`
    - [x] `docs/SUMMARY.md` if it's a new connector
    - [x] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [x] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md` contains a reference to the new connector
    - [x] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified

### CODE REVIEW

**Comment by dmateusp** [comment]
Proof acceptance test passed

**Comment by marcosmarxm** [comment]
thanks @danieldiamond I'm going to review later today and apply tests on it

**Comment by danieldiamond** [comment]
Wrong daniel @marcosmarxm but awesome work @dmateusp 👍

File: airbyte-integrations/connectors/source-us-census/source_us_census/spec.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "documentationUrl": "https://docs.airbyte.io/integrations/sources/uscensus",
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-us-census/source_us_census/spec.json:2]: Nit: The URL should be `https://docs.airbyte.io/integrations/sources/us-census`, just to be consistent with others and slightly more readable.

Our code generator will take care of this part in the future.

**Review by tuliren — APPROVED** [review]
Looks great!

Would you minding adding a connector documentation (`docs/integrations/sources/us-census.md`) based on this template?

https://docs.airbyte.io/contributing-to-airbyte/templates/integration-documentation-template

**Comment by dmateusp** [comment]
hey folks, thanks for the feedback. I tried out the source from the UI yesterday evening and saw some issues I'll have to work on before merging

I'll also address the PR feedback

**Comment by sherifnada** [comment]
@dmateusp let us know if there is anything we can do to help!

**Comment by dmateusp** [comment]
I had a silly issue in the spec and now I fixed the documentation links too :)

**Comment by sherifnada** [comment]
@dmateusp Thanks! we'll review very soon 💪🏼

**Review by sherifnada — APPROVED** [review]

**Pull request merged**